src/diagnostics/MOM_spatial_means.F90

```diff
@@ -350,36 +349,38 @@ function global_mass_integral(h, G, GV, var, on_PE_only, scale, tmp_scale, unsca
   integer :: i, j, k, is, ie, js, je, nz
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
 
-  temp_scale = 1.0 ; if (present(tmp_scale)) temp_scale = tmp_scale
-  scalefac = G%US%L_to_m**2*temp_scale
+  if (present(tmp_scale)) then
+    temp_scale = G%US%RZL2_to_kg * tmp_scale
+    scalefac = 1.0
+  else
+    temp_scale = 1.0
+    scalefac = G%US%RZL2_to_kg
+  endif
   if (present(unscale)) then ; scalefac = scalefac * unscale
   elseif (present(scale)) then ; scalefac = scalefac * scale ; endif
-  tmpForSumming(:,:) = 0.0
 
+  tmpForSumming(:,:) = 0.0
   if (present(var)) then
     do k=1,nz ; do j=js,je ; do i=is,ie
       tmpForSumming(i,j) = tmpForSumming(i,j) + var(i,j,k) * &
-                ((GV%H_to_kg_m2 * h(i,j,k)) * (scalefac*G%areaT(i,j) * G%mask2dT(i,j)))
+                ((GV%H_to_RZ * h(i,j,k)) * (scalefac*G%areaT(i,j) * G%mask2dT(i,j)))
     enddo ; enddo ; enddo
   else
     do k=1,nz ; do j=js,je ; do i=is,ie
       tmpForSumming(i,j) = tmpForSumming(i,j) + &
-                ((GV%H_to_kg_m2 * h(i,j,k)) * (scalefac*G%areaT(i,j) * G%mask2dT(i,j)))
+                ((GV%H_to_RZ * h(i,j,k)) * (scalefac*G%areaT(i,j) * G%mask2dT(i,j)))
```

Comment:
Note that the unit of `tmpForSumming` depends on the presence of `tmp_scale` and `scale`/`unscale`.

Reply:
Well spotted.  The comments have been modified accordingly.